lib/ansible/modules/network/ios/ios_acl.py

```diff
@@ -0,0 +1,1430 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for ios_acl
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+
+DOCUMENTATION = """
+---
+module: ios_acl
+version_added: '2.10'
+short_description: Manages named or numbered ACLs on IOS devices.
+description: This module configures and manages the named or numbered ACLs on IOS platforms.
+author: Sumit Jaiswal (@justjais)
+notes:
+- Tested against Cisco IOSv Version 15.2 on VIRL
+- This module works with connection C(network_cli).
+  See L(IOS Platform Options,../network/user_guide/platform_ios.html).
+options:
+  config:
+    description: A dictionary of ACL options.
+    type: list
+    elements: dict
+    suboptions:
+      afi:
+        description:
+          - The Address Family Indicator (AFI) for the Access Control Lists (ACL).
+        required: true
+        type: str
+        choices:
+          - ipv4
+          - ipv6
+      acls:
+        description:
+          - A list of Access Control Lists (ACL).
+        type: list
+        elements: dict
+        suboptions:
+          name:
+            description: The name or the number of the ACL.
+            required: true
+            type: str
+          acl_type:
+            description:
+              - ACL type
+              - Note, it's mandatory and required for Named ACL, but for
+                Numbered ACL it's not mandatory.
+            type: str
+            choices:
+              - extended
+              - standard
+          aces:
+            description: The entries within the ACL.
+            elements: dict
+            type: list
+            suboptions:
+              grant:
+                description: Specify the action.
+                type: str
+                choices:
+                  - permit
+                  - deny
+              sequence:
+                description:
+                - Sequence Number for the Access Control Entry(ACE).
+                - Refer to vendor documentation for valid values.
+                type: int
+              protocol_options:
```

Comment:
I can't seem to find the `protocol` key in the argspec. As decided we'd have a top-level `protocol` key that would determine the specified protocol for a particular ACE and then a `protocol_options` key, which is essentially a dictionary of dictionaries, where the immediate sub-keys would be the named protocols and the child keys for those sub-keys would be the options we can match.

Ref: `protocol` key - https://github.com/ansible/ansible/pull/66207/files#diff-fa14da51ccc1047e3647da9c0179586fR697
Ref: `protocol_options` key - https://github.com/ansible/ansible/pull/66207/files#diff-fa14da51ccc1047e3647da9c0179586fR706

Reply:
+1

Reply:
On, the side notes to remove slight redundancy IMO if protocol_option is specified, it should not be required that user mentions input to protocol.

Reply:
Yep. I think that's a good idea. We can make them mutually exclusive.